src/Microsoft.AspNetCore.Mvc.Razor/RazorPageActivator.cs

```diff
@@ -33,7 +33,7 @@ public class RazorPageActivator : IRazorPageActivator
             IModelMetadataProvider metadataProvider,
             IUrlHelperFactory urlHelperFactory,
             IJsonHelper jsonHelper,
-            DiagnosticSource diagnosticSource,
+            DiagnosticListener diagnosticListener,
```

Comment:
>DiagnosticListener diagnosticListener, [](start = 12, length = 38)

This is a place where I think we'd like to avoid a breaking change if possible. This is something I expect folks have subclassed.

Reply:
You mean `RazorPageActivator` subclassing?